File: 125-valid-palindrome/125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        newStr = ""
        for c in s:
            if c.isalnum():
                newStr = newStr + c
        left = 0
        right = len(newStr) - 1
        while left <= right:
            if newStr[left] == newStr[right]:
                left += 1
                right -= 1
            else:
                return False
        return True
        

File: 125-valid-palindrome/125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder str = new StringBuilder();
        for (char c : s.toCharArray()) {
            if (Character.isAlphabetic(c) || Character.isDigit(c)) str.append(Character.toLowerCase(c));
        }
        String filteredStr = str.toString();
        int left = 0;
        int right = filteredStr.length() - 1;
        while (left <= right) {
            if (filteredStr.charAt(left) != filteredStr.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}